src/test/java/com/fasterxml/jackson/core/read/NonStandardNumberParsingTest.java

```diff
@@ -31,6 +32,22 @@ public void testLeadingDotInDecimal() throws Exception {
         }
     }
 
+    /*
+     * The format "+NNN" (as opposed to "NNN") is not valid JSON, so this should fail
+     */
+    public void testLeadingPlusSignInDecimal() throws Exception {
+        for (int mode : ALL_MODES) {
+            JsonParser p = createParser(mode, " +123 ");
+            try {
+                p.nextToken();
+                fail("Should not pass");
+            } catch (JsonParseException e) {
+                verifyException(e, "expected digit (0-9) to follow minus sign, for valid numeric value");
```

Comment:
@cowtowncoder I've left this in draft because the existing code throws an exception with a confusing message if a plus sign precedes a number - is it ok if I use this PR to change the message to say that plus signs are not allowed in front of JSON numbers?

Reply:
Yes, absolutely, improvements to error messages are good.